tests/test_deploy.py

```diff
@@ -27,21 +27,31 @@
 from inmanta import deploy
 
 
-def test_deploy(snippetcompiler, tmpdir, postgres_db):
+@pytest.mark.parametrize("default_main_file", [True, False])
+def test_deploy(snippetcompiler, tmpdir, postgres_db, default_main_file):
+    """
+    Test the deploy command. The default_main_file parameter checks that the deploy command accepts
+    files other than `main.cf` through its `-f` cli option.
+    """
     file_name = tmpdir.join("test_file")
     # TODO: when agentconfig deploys no longer require an agent restart, define a new agent. Currently this makes the
     # test to slow.
-    snippetcompiler.setup_for_snippet(
-        """
-    host = std::Host(name="internal", os=std::linux)
-    file = std::Symlink(host=host, source="/dev/null", target="%s")
+    code = f"""
+        host = std::Host(name="internal", os=std::linux)
+        file = std::Symlink(host=host, source="/dev/null", target="{file_name}")
     """
-        % file_name
-    )
+
+    main_file = "main.cf"
+
+    if default_main_file:
+        project = snippetcompiler.setup_for_snippet(code)
+    else:
+        main_file = "other.cf"
+        project = snippetcompiler.setup_for_snippet(code, main_file=main_file)
```

Comment:
Just a simplification:
```suggestion
    main_file = "main.cf" if default_main_file else "other.cf"
     project = snippetcompiler.setup_for_snippet(code, main_file=main_file)
```

Reply:
I wanted to test that the default was being used (i.e. not explicitly providing 'main.cf') but maybe it's not that relevant